container/containerd/handler_test.go

```diff
@@ -42,6 +44,36 @@ func (m *mockedMachineInfo) GetVersionInfo() (*info.VersionInfo, error) {
 	return &info.VersionInfo{}, nil
 }
 
+// Copied from https://github.com/containerd/containerd/blob/main/pkg/protobuf/any.go
+// FromAny converts typeurl.Any to github.com/containerd/containerd/protobuf/types.Any.
+//
+// TODO: vendor these functions once they are moved to github.com/containerd/typeurl.
+//
+//	Currently they are copied over so as not to introduce a dependency on containerd/containerd again
+func FromAny(from typeurl.Any) *anypb.Any {
+	if from == nil {
+		return nil
+	}
+
+	if pbany, ok := from.(*anypb.Any); ok {
+		return pbany
+	}
+
+	return &anypb.Any{
+		TypeUrl: from.GetTypeUrl(),
+		Value:   from.GetValue(),
+	}
+}
+
+// MarshalAnyToProto converts an arbitrary interface to github.com/containerd/containerd/protobuf/types.Any.
+func MarshalAnyToProto(from interface{}) (*anypb.Any, error) {
+	anyType, err := typeurl.MarshalAny(from)
+	if err != nil {
+		return nil, err
+	}
+	return FromAny(anyType), nil
+}
```

Comment:
Can these two functions be unexported, please?

Reply:
There is a PR in https://github.com/containerd/typeurl/pull/45, so that these funcs can be vendored and used instead of copying it over here.